api/retryable.go

```diff
@@ -18,6 +19,23 @@ var retrableErrorSuffixes = []string{
 	io.EOF.Error(),
 }
 
+var retryableStatuses = []int{
+	http.StatusInternalServerError, // 500
+	http.StatusBadGateway,          // 502
+	http.StatusServiceUnavailable,  // 503
+	http.StatusGatewayTimeout,      // 504
+}
+
+// IsRetryableStatus returns true if the response's StatusCode is one that we should retry.
+func IsRetryableStatus(r *Response) bool {
+	for _, status := range retryableStatuses {
+		if r.StatusCode == status {
+			return true
+		}
+	}
+	return false
```

Comment:
[super nit] we have the magic of generics now, so we could replace this with
```suggestion
	slices.Contains(retryableStatuses, status)
```

if you really wanted to, but this is perfectly fine as is.

Reply:
:+1: I've embraced the magic of generics in the latest commit.

Reply:
✨